Lotus docs restructuring

LexLuthr opened the pull request:
Fixes https://github.com/filecoin-project/lotus-docs/issues/64

This PR is still under progress. Please do not merge it before consulting with the author.

Steps:

- [x] Move the files around - @LexLuthr 

- [x] Create new files as per the mind map - @LexLuthr 

- [x] Create the new layout - @LexLuthr 

- [x] Add KB section - @LexLuthr 

- [x] Start populating the new files - @LexLuthr

- [x] Move around the data in the older files - @LexLuthr 

- [x] Verify all links are working - @LexLuthr 

- [x] Test the new docs website

johnnymatthews commented:
I took a look at this PR/branch and found out why the sidebar is freaking out. Essentially, the menu items in `/config/_default/menus/menus.en.toml` don't match what's in `/content/en/docs`

![image](https://user-images.githubusercontent.com/9611008/151893244-e41f3d90-f6f5-4327-a268-04f265252a4c.png)

We'll need to move pages into separate topbar sections and then into their sidebar sections. For example, Lotus and Storage providers need to have their own topbar section, according to the [Miro board](https://miro.com/app/board/uXjVOWo-d_E=/).

However, I'm hesitant to do this while you're still populating the folders within this branch. I'm going to wait until you've finished creating each page. Once you've done that, I'll start building the main (topbar) sections. While I'm doing that, you can create the list of redirects we'll need to make sure that users don't hit 404 pages.

LexLuthr commented:
> I took a look at this PR/branch and found out why the sidebar is freaking out. Essentially, the menu items in `/config/_default/menus/menus.en.toml` don't match what's in `/content/en/docs`
> 
> ![image](https://user-images.githubusercontent.com/9611008/151893244-e41f3d90-f6f5-4327-a268-04f265252a4c.png)
> 
> We'll need to move pages into separate topbar sections and then into their sidebar sections. For example, Lotus and Storage providers need to have their own topbar section, according to the [Miro board](https://miro.com/app/board/uXjVOWo-d_E=/).
> 
> However, I'm hesitant to do this while you're still populating the folders within this branch. I'm going to wait until you've finished creating each page. Once you've done that, I'll start building the main (topbar) sections. While I'm doing that, you can create the list of redirects we'll need to make sure that users don't hit 404 pages.

Hey @johnnymatthews , the place holders are all in place. New files only need to be populated now. I think we are ready to restructure the top menu and side menu. In case there are any confusion on the Miro board, I am listing out the top menu below.

1. Lotus - Project would be nested under here
2. Storage Providers
3. Tutorials
4. Developers - APIs will be nested under here
5. CLI

johnnymatthews commented:
> Lotus - Project would be nested under here
> Storage Providers
> Tutorials
> Developers - APIs will be nested under here
> CLI

This is super useful, thanks!

johnnymatthews commented on config/_default/menus/menus.en.toml:
```diff
@@ -1,54 +1,105 @@
+
+[[main]]
+  name = "Knowledge Base"
+  url = "/kb"
+  identifier = "kb"
```
```suggestion
  url = "/kb"
  identifier = "kb"
```

Let's change this to `knowledge-base` on the off-chance that someone mistakes it for kilobytes.

LexLuthr replied:
@johnnymatthews Can we keep it for now? I will open another issue and work on changing this. I would prefer to get it merged as it is blocker for lotus.

johnnymatthews replied:
Ok, sure.

johnnymatthews commented on content/en/lotus/lotus-config/configuration.md:
```diff
@@ -0,0 +1,273 @@
+Variables common to most Lotus binaries:
+
+- `LOTUS_FD_MAX`: Sets the file descriptor limit for the process
+- `LOTUS_JAEGER`: Sets the Jaeger URL to send traces. See TODO.
```
> See TODO.

Where's this meant to go?

LexLuthr replied:
It is carried forwarded from older content. I did not add it. Shall we remove it?
I think someone wanted to add Jaeger URL maybe?

johnnymatthews replied:
Yeah let's remove it for now. Thanks.

LexLuthr replied:
I have filed the latest commit to remove this TODO. I have also hidden the blog and empty miner tutorials. Reload your browser from source to avoid cached version and you should see the changes.

johnnymatthews reviewed (changes requested):
This looks good, there's just a couple changes requested. Also, how are redirects being managed?

johnnymatthews commented:
1. The top-border is missing on the topbar nav on the homepage:

    ![image](https://user-images.githubusercontent.com/9611008/154142367-3cf8ce4c-88f3-4f0b-94bc-e2288446e544.png)

1. We need some way of identifying which section the user is in. Bolding the topbar nav menu item would probably work:

    ![image](https://user-images.githubusercontent.com/9611008/154142574-ad713a39-be29-4a58-ad04-e2319c67b87b.png)

LexLuthr commented:
> 1. The top-border is missing on the topbar nav on the homepage:
>    ![image](https://user-images.githubusercontent.com/9611008/154142367-3cf8ce4c-88f3-4f0b-94bc-e2288446e544.png)

I had removed in a previous commit but I have put it back. I think your browser is caching the files. Please reload from source to check this once.

> 2. We need some way of identifying which section the user is in. Bolding the topbar nav menu item would probably work:
>    ![image](https://user-images.githubusercontent.com/9611008/154142574-ad713a39-be29-4a58-ad04-e2319c67b87b.png)

I think your browser is caching the files. Top Menu item should change color when user is in a section. Please reload from source to check this once.

> 3. The **Lotus Miner Tutorials** dropdown under the **Tutorials** section doesn't do anything.

We will be adding tutorials here. If this PR gets merged before Friday, then we will be releasing first miner tutorial on snapdeals before Thursday next week.
If you would prefer to hide it for now, I can.

> 4. Remove the **Blog** section for now.

I am planning to run a bounty program to add blogs and articles. I will hide it for now till at least 1 real blog is in place. 

> 5. **Filecoin Economics** disappears if you're not on the **What is Lotus** page:
>    ![image](https://user-images.githubusercontent.com/9611008/154144452-a9d452b3-2992-4bcc-ba79-b2059b851e25.png)

This again sounds like a caching problem on your browser. I have removed that page for now. I will be adding that later only as Fil plus page.

> 6. **Client** might be a better section title, rather than **Lotus**. Changing this would mean that **Lotus Tutorials** would become **Client tutorials**, which makes more sense.

I have been avoiding calling it client so people don't start confusing it with a web 2.0 model. Calling it client is also unfair as it does more than just being a client. Every SP needs a lotus node and SP section refers back to Lotus section multiple times. This would again confuse people if we call it a client. @jennijuju Can you weigh in here? I really don't want to change names as it would be very confusing for the community. 

I will wait for your answer on these point for the final commit.

LexLuthr commented:
> This looks good, there's just a couple changes requested. Also, how are redirects being managed?

Can you clarify what redirects?  Because I just hacked my way through and did not develop this.

LexLuthr commented:
@johnnymatthews I think with my latest commit, we are good to merge apart from the lotus/client terminology part. I have already explained the rest of the stuff in my previous comments.

johnnymatthews commented:
The preview on this PR faulted for some reasons (likely because the Netlify site was deleted). Closed this PR in favour of https://github.com/filecoin-project/lotus-docs/pull/82. Let's carry on the discussion there.
